apps/namadillo/src/slices/proposals/functions.ts

```diff
@@ -308,71 +351,33 @@ export const fetchVotedProposalIds = async (
   return proposalIds;
 };
 
-export const performVote = async (
-  api: DefaultApi,
+export const createVoteProposalTx = async (
   proposalId: bigint,
   vote: VoteType,
   account: Account,
+  gasConfig: GasConfig,
   chain: Chain
-): Promise<void> => {
-  const signingClient = getIntegration("namada").signer();
-
-  if (typeof signingClient === "undefined") {
-    throw new Error("no signer");
-  }
-
-  const publicKey = account.publicKey;
-  if (typeof publicKey === "undefined") {
-    throw new Error("no public key on account");
-  }
-  const { tx, rpc } = await getSdkInstance();
-  const signer = account.address;
-
-  const proposalProps = {
-    signer,
-    proposalId,
-    vote,
-  };
-
-  const wrapperTxProps = {
-    token: nativeToken,
-    feeAmount: BigNumber(0),
-    gasLimit: BigNumber(20_000),
-    chainId: chain.chainId,
-    publicKey,
-    memo: "",
-  };
-
-  const txArray: EncodedTx[] = [];
-
-  // RevealPK if needed
-  const { publicKey: pk } = (await api.apiV1RevealedPublicKeyAddressGet(signer))
-    .data;
-
-  if (!pk) {
-    const revealPkTx = await tx.buildRevealPk(wrapperTxProps, signer);
-    // Add to txArray to sign & broadcast below:
-    txArray.push(revealPkTx);
-  }
-
-  const encodedTx = await tx.buildVoteProposal(wrapperTxProps, proposalProps);
-  txArray.push(encodedTx);
-
+): Promise<TransactionPair<VoteProposalProps>[]> => {
   try {
-    const signedTx = await signingClient.sign(
-      signer,
-      txArray.map(({ tx }) => tx)
+    const { tx } = await getSdkInstance();
+
+    const voteProposalProps = {
+      signer: account.address,
+      proposalId: BigInt(1),
```

Comment:
Not sure if this is intended, because we have `proposalId` param unused

Reply:
Oh nice spot! That is definitely a mistake. I've fixed it and added a commit which will make the linter catch this error.